src/main/scala/streamit/client/ApacheKafkaClient.scala

```diff
@@ -0,0 +1,257 @@
+package streamit.client
+
+import java.time.{ Instant, ZoneId, ZonedDateTime, Duration => JavaDuration }
+import java.util.{ Properties, Collection => JavaCollection }
+
+import cats.effect._
+import cats.syntax.apply._
+import fs2.Stream
+import fs2.kafka.{ KafkaConsumer => _ }
+import io.confluent.kafka.serializers.AbstractKafkaAvroSerDeConfig
+import log.effect.LogWriter
+import org.apache.kafka.clients.admin.{ AdminClient, AdminClientConfig }
+import org.apache.kafka.clients.consumer._
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.serialization.Deserializer
+import org.apache.zookeeper.ZooKeeper
+import org.slf4j.LoggerFactory
+import streamit._
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+
+object ApacheKafkaClient {
+  def apply[F[_]: ConcurrentEffect: LogWriter](kafkaSettings: KafkaSettings): ApacheKafkaClient[F] =
+    new ApacheKafkaClient(kafkaSettings)
+}
+
+/**
+  * While ideally an fs2-first client such as the fs2-kafka client would be the only one in use by the
+  * KafkaTaskRunner, it is quite limited (especially in documentation), so when we need to do lower level
+  * consumergroup manipulations, we need to resort to the standard java client, so that's all being
+  * isolated to this client.
+  */
+class ApacheKafkaClient[F[_]: ConcurrentEffect](settings: KafkaSettings)(
+  implicit logger: LogWriter[F]
+) {
+
+  // WARNING: Here be dragons.  Or at least, sphincter-tightening jumps between FP and non-FP with
+  // ugly java conversions. All help at making this less ugly, very, very welcome.
+
+  // kafka API listeners etc aren't in F[_] so we can't use 'logger' :(
+  private[this] val slfLogger = LoggerFactory.getLogger("ApacheKafkaClient")
+
+  /**
+    * @return A [[fs2.Stream]] containing a [[org.apache.zookeeper.ZooKeeper]] client, which will be
+    *         auto-closed via Stream.bracket.
+    */
+  def createZookeeperClient(): Stream[F, ZooKeeper] =
+    settings.zookeeper match {
+      case None =>
+        // zookeeper is an optional config, even if the kafkarunner gets configured
+        Stream
+          .eval(logger.warn("Zookeeper host not configured, skipping ZK test"))
+          .flatMap(_ => Stream.empty)
+      case Some(zkHost) =>
+        Stream.bracketCase(
+          ConcurrentEffect[F].delay(new ZooKeeper(zkHost, 10000, null)) <*
+            logger.info(s"Created zookeeper client to talk to zk host: $zkHost")
+        )(
+          (ac, _) =>
+            ConcurrentEffect[F].delay(ac.close()) *> logger.info(s"Zookeeper client Closed")
```

Comment:
Considering this https://github.com/typelevel/cats-effect/issues/401 we might use `>>` in these cases. We just need `import cats.syntax.flatMap._` and it's equivalent here anyway.
```suggestion
            ConcurrentEffect[F].delay(ac.close()) >> logger.info(s"Zookeeper client Closed")
```

Reply:
done